### PR TITLE
Refactors OSX menu logic

### PR DESCRIPTION
**Pull request by naps62**

### CODE REVIEW

**Review by gabrielpoca — APPROVED** [review]
👍 makes sense

**Comment by naps62** [comment]
@gabrielpoca can you run it locally to make sure I didn't break anything on OSX?

File: bin/iron/src/app.rs [hunk 1]
```diff
@@ -3,24 +3,19 @@ use std::path::PathBuf;
...
-use tauri::{AboutMetadata, Menu, MenuItem, Submenu, WindowMenuEvent};
 use tauri::{
-    AppHandle, Builder, CustomMenuItem, GlobalWindowEvent, Manager, SystemTray, SystemTrayEvent,
-    SystemTrayMenu, SystemTrayMenuItem, WindowBuilder, WindowEvent, WindowUrl,
+    AppHandle, Builder, GlobalWindowEvent, Manager, WindowBuilder, WindowEvent, WindowUrl,
 };
 use tauri_plugin_window_state::{AppHandleExt, Builder as windowStatePlugin, StateFlags};
 
-use crate::{commands, error::AppResult};
+use crate::{commands, error::AppResult, menu, system_tray};
```
Comment (gabrielpoca) [inline bin/iron/src/app.rs:11]: Ok, this fails beacuse you dont' have `system_tray` in macos: https://github.com/iron-wallet/iron/pull/351/files#diff-19ee4c60c2c483915400021b6e5b2e6d03f82e230f1b0c3112a44d33812fe5d9R8

Comment (naps62) [reply bin/iron/src/app.rs]: updated. can you try again?

I was trying to build for macos myself, but apparently can't be done on linux

File: bin/iron/src/menu.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+use tauri::{Menu, WindowMenuEvent};
+
+#[cfg(not(target_os = "macos"))]
+pub(crate) fn build() -> Menu {
+    Default::default()
+}
+
+#[cfg(target_os = "macos")]
+fn build() -> Menu {
```
Comment (gabrielpoca) [inline bin/iron/src/menu.rs:9]: just missing a "pub" here and that's it

**Pull request merged**